app/models/job.js

```diff
@@ -140,8 +141,14 @@ export default Model.extend(DurationCalculations, {
       return;
     }
     this.set('subscribed', true);
-    if (Travis.pusher) {
-      return Travis.pusher.subscribe('job-' + (this.get('id')));
+
+    if (Travis.pusher && Travis.pusher.ajaxService) {
+      return Travis.pusher.ajaxService.post(Pusher.channel_auth_endpoint, {
+        socket_id: Travis.pusher.pusherSocketId,
+        channels: ['private-job' + this.get('id')]
+      }).then(() => {
+        return Travis.pusher.subscribe('job-' + (this.get('id')));
+      });
```

Comment:
Totally agree that this should be in its own service.  That said, I started going down the road of extracting this and started to run into issues and figured it would be better to get it out in its current state.